### PR TITLE
feat: Moved GitHub Projects API usage from REST to GraphQL

### LINKED ISSUE #249: GitHub projects REST API move to GraphQL
### Description

In my Drifty project, I use GitHub projects for streamlining issues and milestones and is actively updated.
But HealthCheck shows that GitHub Projects is not used and is asked to be removed.

You can check the latest report generated here - https://healthcheck.eddiehubcommunity.org/repo/report/cm0ze3dl800151w3l05qyxazu

### Screenshots

image

### Additional information

I wish to discuss and work on this issue.

### PR DESCRIPTION
**Pull request by SaptarshiSarkar12**
## Fixes Issue
Closes #249

## Changes proposed

This PR makes the following changes:
- Adds support for GitHub Projects V2 via GraphQL migrating from REST API

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot

[Screen Recording of my changes](https://github.com/user-attachments/assets/4b77587d-7922-4543-b095-45ea6cedfd7e)

## Note to reviewers

I could not find a way to fetch only `OPENED` projects and, I had to fetch `first 10`. Further, I requested only `id`, `title` and `closed`. I filtered those data in terms of `!closed` in the `ProjectsCheck` section. There is a rare case that all the 10 are closed. Let me know what I should do in that case. I have one option to try a `while` or `for` loop but that would soon reach API limit.

### CODE REVIEW

File: src/utils/github/getProjectsApi.js [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+        }
+      }
+      `,
+      {
+        owner,
+        repo,
+      },
+    );
+    console.log(response);
+    console.error(response);
```
Comment (eddiejaoude) [inline src/utils/github/getProjectsApi.js:36]: I don't think this debug code is required anymore

```suggestion

```

File: src/utils/github/getProjectsApi.js [hunk 2]
```diff
@@ -1,28 +1,51 @@
...
+      data: response.repository.projectsV2.nodes,
+    };
   } catch (e) {
     console.error(e);
     response = {
       status: 404,
       data: [],
     };
   }
+  console.log(response);
```
[the remark this answers is not in the document]
Comment (SaptarshiSarkar12) [reply src/utils/github/getProjectsApi.js]: Yeah, forgot to remove those 😂

File: src/utils/github/getProjectsApi.js (hunk 1 above)
Comment (SaptarshiSarkar12) [reply src/utils/github/getProjectsApi.js]: True 😅

**Comment by SaptarshiSarkar12** [comment]
@eddiejaoude Thank you Eddie for reviewing and merging my PR 😁! I got to learn more about GraphQL and how GitHub integrates it in its API. 
Thank you for the opportunity to contribute 😄.

**Comment by eddiejaoude** [comment]
I learnt about GraphQL from your PR also 🚀 so double thank you 🎉

**Comment by SaptarshiSarkar12** [comment]
@eddiejaoude Most welcome 😁. Grateful to Open-Source.

**Pull request merged**